From b49c892faa59aa457b59b69b89be59339d9b41bc Mon Sep 17 00:00:00 2001
From: Camille <onodera@openmailbox.org>
Date: Wed, 27 Jan 2016 22:01:48 +0100
Subject: [PATCH] Update screen.C

---
 src/screen.C | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/src/screen.C b/src/screen.C
index 9eb375a..412de27 100644
--- a/src/screen.C
+++ b/src/screen.C
@@ -2443,8 +2443,6 @@ rxvt_term::scr_refresh () NOTHROW
    */
   if (showcursor)
     {
-      if (focus)
-        {
           if (cursor_type == 0)
             scr_set_char_rend (ROW(screen.cur.row), cur_col, cur_rend);
           else if (oldcursor.row >= 0)
@@ -2460,20 +2458,10 @@ rxvt_term::scr_refresh () NOTHROW
                 XFillRectangle (dpy, vt, gc,
                                 Col2Pixel (cur_col),
                                 Row2Pixel (oldcursor.row),
-                                2,
+                                1,
                                 Height2Pixel (1));
             }
-        }
-      else if (oldcursor.row >= 0)
-        {
-          XSetForeground (dpy, gc, pix_colors[ccol1]);
 
-          XDrawRectangle (dpy, vt, gc,
-                          Col2Pixel (cur_col),
-                          Row2Pixel (oldcursor.row),
-                          (unsigned int) (Width2Pixel (cursorwidth) - 1),
-                          (unsigned int) (Height2Pixel (1) - 1));
-        }
     }
 
   /*
